tests/lib-tests.nix

```
@@ -374,6 +374,28 @@ let
         ];
       };
     };
+
+    testEscapeStringForLua = {
+      expr = {
+        simple = helpers.utils.escapeStringForLua "simple";
+        depth-one = helpers.utils.escapeStringForLua "]]";
+      };
+      expected = {
+        simple = "[[simple]]";
+        depth-one = "[=[]]]=]";
+      };
+    };
+
+    testEscapeStringForVimscript = {
+      expr = {
+        simple = helpers.utils.escapeStringForVimscript "simple";
+        depth-one = helpers.utils.escapeStringForVimscript "EOF";
+      };
+      expected = {
+        simple = "<< EOF\nsimple\nEOF";
+        depth-one = "<< EOFF\nEOF\nEOFF";
```

We can use "indented strings" to make this more legible:

```suggestion
        simple = ''
          << EOF
          simple
          EOF'';
        depth-one = ''
          << EOFF
          EOF
          EOFF'';
```

Note: while it's nice to have the closing `''` on its own line, this will add a trailing `\n` to the string.